source/synthDrivers/espeak.py

```diff
@@ -90,8 +99,69 @@ def _processText(self, text):
 			0x5B: u" [", # [: [[ indicates phonemes
 		})
 
-	def speak(self,speechSequence):
-		defaultLanguage=self._language
+	def _determineLangFromCommand(self, command: LangChangeCommand) -> Optional[str]:
+		"""
+		Checks if a lang code given is compatible with eSpeak.
+		If not, check if a default mapping occurs in L{_defaultLangToLocale}.
+		Otherwise, checks if a language of a different dialect exists (e.g. ru-ru to ru).
+		Returns an eSpeak compatible lang code or None, which should be handled as
+		falling back to the default language.
+		"""
+		# Use default language if no command.lang is supplied
+		langWithLocale = command.lang if command.lang else self._language
+		langWithLocale = langWithLocale.lower().replace('_', '-')
+
+		langWithoutLocale: Optional[str] = langWithLocale.split('-')[0]
+
+		# Check for any language where the language code matches, regardless of dialect: e.g. ru-ru to ru
+		matchingLanguages = filter(lambda l: l.split('-')[0] == langWithoutLocale, self.availableLanguages)

```

Comment:
Please don't use single character variable names, especially ones like `l` which can be easily confused with `I` or even `1`. 
```suggestion
		matchingLanguages = filter(
			lambda lang: lang.split('-')[0] == langWithoutLocale, self.availableLanguages
		)

```

Reply:
Fixed in 5c3388b